src/index.js

```diff
@@ -1,31 +1,81 @@
 import http from 'http';
+import path from 'path';
 import bodyParser from 'body-parser';
 import cors from 'cors';
 import express from 'express';
 import winston from 'winston';
+import yamlConfig from 'node-yaml-config';
+import passport from 'passport';
+import mongoose from 'mongoose';
+import FacebookStrategy from 'passport-facebook';
 
 import api from './api';
-import config from './config.json';
+import facebook from './auth/facebook';
 import middleware from './middleware';
+import User from './db/models/user';
+
+const config = yamlConfig.load(path.join(__dirname, '/config.yml'));
+
+const mongodbUrl = `mongodb://${config.database.host}:${config.database.port}/${config.database.db}`;
+
+winston.log(`Connecting to mongodb at: ${mongodbUrl}`);
+
+mongoose.connect(mongodbUrl);
+
+winston.log('info', config);
+// console.log('info', config);
+
+passport.use(new FacebookStrategy({
+  clientID: config.facebook.appId,
+  clientSecret: config.facebook.secret,
+  callbackURL: config.facebook.callbackUrl,
+  display: 'popup',
+  authType: 'code',
+  passReqToCallback: 'true',
+  enableProof: true,
+}), (request, accessToken, refreshToken, profile, callback) => {
+  User.findOne({ oauthId: profile.id }, (error, user) => {
```

Comment:
Change this to "promise pattern" with `then` and `error` http://mongoosejs.com/docs/promises.html. We should also consider adding Bluebird.